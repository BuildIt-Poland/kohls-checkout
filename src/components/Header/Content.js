import styled from 'styled-components';
import ContentBase from '../Content';

const Content = styled(ContentBase)`
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  height: 100%;
  padding: 0;
`;

export default Content;
