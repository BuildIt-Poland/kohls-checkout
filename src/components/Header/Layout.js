import styled from 'styled-components';

import { colorLightGrey, borderWidthThin, colorWhite } from '../../styles/designTokens';

const Layout = styled.header`
  position: fixed;
  width: 100%;
  height: 6rem;
  background-color: ${colorWhite};
  border-bottom: ${borderWidthThin} solid ${colorLightGrey};
  top: 0;
`;

export default Layout;
