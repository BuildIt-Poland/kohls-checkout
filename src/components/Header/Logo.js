import React from 'react';
import { Link as LinkBase } from 'react-router-dom';
import styled from 'styled-components';

import { Logo as LogoIcon } from '../Icons';

const Link = styled(LinkBase)`
  height: 100%;
  cursor: pointer;
  padding-top: 2rem;
  position: relative;
`;

const AccessibleText = styled.span`
  position: absolute;
  left: -99999rem;
`;

function Logo() {
  return (
    <Link data-testid="logo" to="/">
      <AccessibleText>The Store</AccessibleText>
      <LogoIcon />
    </Link>
  );
}

export default Logo;
