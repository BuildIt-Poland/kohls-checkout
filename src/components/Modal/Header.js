import styled from 'styled-components';

import { spacingMedium, colorLightGrey, borderWidthThin, spacingSmall, fontLarge } from '../../styles/designTokens';

const Header = styled.h3`
  height: 6rem;
  font-size: ${fontLarge};
  border-bottom: ${borderWidthThin} solid ${colorLightGrey};
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0;
  padding: 0 ${spacingSmall} 0 ${spacingMedium};
  text-align: left;
  line-height: 1;
`;

export default Header;
