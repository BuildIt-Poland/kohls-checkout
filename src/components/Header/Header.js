// Default website header
import React from 'react';

import CartIcon from '../CartIcon';
import Layout from './Layout';
import Content from './Content';
import Logo from './Logo';

function Header() {
  return (
    <Layout>
      <Content>
        <Logo />
        <CartIcon />
      </Content>
    </Layout>
  );
}

export default Header;
