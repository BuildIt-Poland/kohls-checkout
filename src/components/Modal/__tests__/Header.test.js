import React from 'react';
import { create } from 'react-test-renderer';

import Header from '../Header';

describe('COMPONENT - Modal Header', () => {
  it('renders correctly', () => {
    const component = create(<Header />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
