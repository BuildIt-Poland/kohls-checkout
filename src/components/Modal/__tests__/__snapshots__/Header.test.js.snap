// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Modal Header renders correctly 1`] = `
.c0 {
  height: 6rem;
  font-size: 2.4rem;
  border-bottom: 1px solid #b0b0b0;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0;
  padding: 0 1rem 0 2rem;
  text-align: left;
  line-height: 1;
}

<h3
  className="c0"
/>
`;
